ONSdigital/census-rh-service
# Cr266 add endpoint to request sms fulfilment

# Motivation and Context


# What has changed
Addition of new endpoint to CaseEndpoint, and corresponding business level code in CaseServiceImpl.
Unit tests added.

# How to test?
Run unit tests.

To manually test:

1. Make sure that you GCP project has Firestore enabled. This should be in 'Native' mode.
2. Load a case. See the RH service readme.md for some notes on how to do this. Make sure that the addressType is set to 'HI'.
3. Confirm that case has been loaded into Firestore.
4. Invoke the new endpoint. On the command line this can be done with HTTPie:
```
$ http POST "http://localhost:8071/cases/dc4477d1-dd3f-4c69-b181-7ff725dc9fa4/fulfilments/sms" caseId=dc4477d1-dd3f-4c69-b181-7ff725dc9fa4 telNo=07714111111 fulfilmentCode=UACIT1 dateTime="2016-11-09T11:44:44.797Z"
```
5. Check the fulfilment request message on a queue bound to 'event.fulfilment.request'.

# Links
See: https://collaborate2.ons.gov.uk/jira/browse/CR-266
